src/server/main_service.cc

```diff
@@ -661,9 +661,7 @@ void Service::Init(util::AcceptServer* acceptor, std::vector<facade::Listener*>
   pp_.Await([reg](uint32_t index, ProactorBase* pb) { ServerState::Init(index, reg); });
 
   uint32_t shard_num = GetFlag(FLAGS_num_shards);
-  if (shard_num == 0) {
-    shard_num = pp_.size() > 1 ? pp_.size() - 1 : pp_.size();
-  } else if (shard_num > pp_.size()) {
```

Comment:
BTW, we might want to `exit(-1)`, or at least print a `WARNING` in this case?

Reply:
warning is fine